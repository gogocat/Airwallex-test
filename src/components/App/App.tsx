import React from 'react';
import CssBaseline from '@mui/material/CssBaseline';
import Header from 'components/Header/Header';
import Footer from 'components/Footer/Footer';
import HeroIntro from 'components/HeroIntro/HeroIntro';

import { 
    createTheme, 
    ThemeProvider
} from '@mui/material/styles';

const defaultTheme = createTheme();

function App() {
    return (
        <ThemeProvider theme={defaultTheme}>
            <CssBaseline />
            <Header />
            <main className='main'>
                <HeroIntro />
            </main>
            <Footer />
        </ThemeProvider>
    );
}

export default App;
