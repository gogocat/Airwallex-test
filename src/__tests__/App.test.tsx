import React from 'react';
import {
    render,
    screen,
} from '@testing-library/react';
import App from 'components/App/App';

describe('App', () => {
    it('should renders with HeroIntro content', () => {
        const {
            asFragment,
        } = render(<App />);

        const header = screen.getByText('BROCCOLI & CO.', { selector: 'header h6'});
        const footer = screen.getByText('Made with â™¥ in Melbourne.', { selector: 'footer p'});
        const introTitle = screen.getByText('A better way to enjoy everyday.');

        expect(header).toBeInTheDocument();
        expect(footer).toBeInTheDocument();
        expect(introTitle).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});