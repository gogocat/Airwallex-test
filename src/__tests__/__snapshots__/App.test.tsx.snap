// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders with HeroIntro content 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative header css-1vazdj7-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-2l1qg7-MuiTypography-root"
      >
        BROCCOLI & CO.
      </h6>
    </div>
  </header>
  <main
    class="main"
  >
    <div
      class="hero-intro MuiBox-root css-10m0cap"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
        >
          A better way to enjoy everyday.
        </h1>
        <p
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-paragraph css-17rfbpd-MuiTypography-root"
        >
          Be the first to know when we launch.
        </p>
        <div
          class="MuiStack-root css-kof1v3-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Request an invite
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </main>
  <footer
    class="footer MuiBox-root css-16ybtzv"
  >
    <p
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter css-1xatcrf-MuiTypography-root"
    >
      Made with ♥ in Melbourne.
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
    >
      © 2023 Broccoli & co. All rights reserved..
    </p>
  </footer>
</DocumentFragment>
`;
